### ISSUE TITLE
`compute`-Parameter for bones to allow for computed values

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-core#629]
> Result of the viur-meeting: This feature might be useful. Let's start a new attempt as a `evaluate`-parameter on BaseBone.

_Originally posted by @phorward in https://github.com/viur-framework/viur-core/issues/623#issuecomment-1398489726_

This issue is a substitution for #623. It is useful to quarantee expected data types and bone rendering behaviors with computed values.

- `BaseBone` with `compute`-parameter as a dataclass
- The dataclass should require for
  - a lambda for value computation (simular to #623)
  - an optional threshold, either with the options
    - compute always
    - compute on add/edit only
    - re-compute when threshold of X seconds has passed
- Bones using the `compute`-flag should at first be limited to
  - `readOnly=True`
  - `multiple=False` - this could be improved later on.
  - `languages=None` - this could be improved later on.

This issue relates to #616 as well.

**Comment by ArneGudermann** [comment]
@viur-framework/developers 
I have a question about the implementation. For example, what do we do if I have the following code. 
```python
def testme():
    return "123.1"

compute_me = NumericBone(
    readOnly=True,
    compute=testme,
    threshold=ThresholdValue(method=ThresholdMethods.Always))
 ```
In this case, should we convert the value we get from `testme` to `int` or should we keep the value as it is?

In general, we should discuss when the `compute` function returns a value that does not match the bone.

**Comment by phorward** [comment]
Hello @ArneGudermann,

I think that this namedtuple + enum definition would generally be the better choice in BaseBone. This draft defines all required features, and might also answer your question: The compute-definition might be configured to allow for raw values or to pass them through the bone's fromClient. 
```python
from enum import Enum, unique

@unique
class ComputeThreshold(Enum):
    ALWAYS = 1
    ONCE = 2
    SECONDS = 3

    def __init__(self, value=None):
        self.value = value

    @classmethod
    def seconds(cls, value):
        instance = cls.SECONDS
        instance._value_ = value
        instance.value = value
        return instance

@dataclass
class Compute:
    fn: callable,  # the callable computing the value
    threshold: ComputeThreshold = ComputeThreshold.ALWAYS   # the value caching threshold
    raw: bool = true  # defines whether the value is used as is, or is passed to bone.fromClient
```
Afterwards, `compute_me` can be defined this way:
```python
compute_me = NumericBone(
    compute=Compute(testme)
)

compute_me_2min = NumericBone(
    compute=Compute(testme, ComputeThreshold.seconds(60 * 2))
)
```
The `readOnly` is automatically enforced by the constructor.

What do you think?

**Comment by phorward** [comment]
Implemented by #639
